<launch>
  <arg name="gui" default="true"/>

  <node name="sample_image_publisher"
        pkg="jsk_recognition_utils" type="sample_image_publisher.py">
    <rosparam subst_value="true">
      image_file: $(find jsk_recognition_utils)/sample/images/real_world_checkerboard_grid_7x6_size_0.108/depth_image_raw.npz
      camera_info_file: $(find jsk_recognition_utils)/sample/images/real_world_checkerboard_grid_7x6_size_0.108/depth_camera_info.yaml
      frame_id: camera_depth_optical_frame
      encoding: 16UC1
      rate: 30.0
    </rosparam>
  </node>

  <node name="image_16uc1_to_32fc1"
        pkg="jsk_recognition_utils" type="image_16uc1_to_32fc1.py">
    <remap from="~input" to="sample_image_publisher/output/image_raw"/>
  </node>

  <group if="$(arg gui)">
    <node name="image_view"
          pkg="image_view" type="image_view">
      <remap from="image" to="image_16uc1_to_32fc1/output"/>
      <rosparam>
        do_dynamic_scaling: true
      </rosparam>
    </node>
  </group>
</launch>
